src/base/bittorrent/session.cpp

```diff
@@ -2152,24 +2151,24 @@ bool Session::addTorrent_impl(const std::variant<MagnetUri, TorrentInfo> &source
 
         const TorrentContentLayout contentLayout = ((loadTorrentParams.contentLayout == TorrentContentLayout::Original)
                                                     ? detectContentLayout(torrentInfo.filePaths()) : loadTorrentParams.contentLayout);
-        QStringList filePaths = (!addTorrentParams.filePaths.isEmpty() ? addTorrentParams.filePaths : torrentInfo.filePaths());
-        applyContentLayout(filePaths, contentLayout, Utils::Fs::findRootFolder(torrentInfo.filePaths()));
+        PathList filePaths = (!addTorrentParams.filePaths.isEmpty() ? addTorrentParams.filePaths : torrentInfo.filePaths());
+        applyContentLayout(filePaths, contentLayout, Path::findRootFolder(torrentInfo.filePaths()));
 
         // if torrent name wasn't explicitly set we handle the case of
         // initial renaming of torrent content and rename torrent accordingly
         if (loadTorrentParams.name.isEmpty())
         {
-            QString contentName = Utils::Fs::findRootFolder(filePaths);
+            QString contentName = Path::findRootFolder(filePaths).data();
```

Comment:
looks like `toString()` since we cannot be certain what processing follows it.

Reply:
> looks like `toString()` since we cannot be certain what processing follows it.

`Path::findRootFolder` returns single-element path. Why to search separators in it?

Reply:
>Why to search separators in it?

I still hold the same viewpoint: `data()` should only be used if the following code is able to consume it. Otherwise it is always `toString()`. I consider it to be safe even if it is sacrificing some performance.